lib/oban/instance.ex

```diff
@@ -0,0 +1,224 @@
+defmodule Oban.Instance do
+  @moduledoc """
+  Provides a simplified interface to the `Oban` library.
+  """
+
+  @doc """
+  It creates a facade for the `Oban` functions.
+
+  It allows you to avoid having to pass the `t:Oban.name/0` value to the `Oban` functions, as it is automatically set to
+  the module name. As well as allowing you to configure the `Oban` instance in the application's configuration using
+  the module name under a given OTP application key.
+
+  ## Examples
+
+  ### In an application module
+
+      defmodule MyApp.Oban do
+        use Oban.Instance,
+          otp_app: :my_app,
+          repo: MyApp.Repo
+      end
+
+  Now you can register the `MyApp.Oban` module in the application's supervision tree:
+
+      defmodule MyApp.Application do
+        use Application
+
+        def start(_type, _args) do
+          children = [
+            MyApp.Repo,
+            MyApp.Oban
+          ]
+
+          opts = [strategy: :one_for_one, name: MyApp.Supervisor]
+          Supervisor.start_link(children, opts)
+        end
+      end
+
+  ### Avoiding the need of passing the `Oban` instance
+
+  Instead of calling `Oban` functions passing the `Oban` instance, you can use the `Oban.Instance` module directly:
+
+      Oban.Instance.insert(MyApp.MyJob.new(args: %{ "field" => "value" }))
+  """
+  defmacro __using__(opts \\ []) do
+    {opts, child_opts} = Keyword.split(opts, [:otp_app])
+    otp_app = Keyword.fetch!(opts, :otp_app)
+
+    quote do
+      @doc """
```

Comment:
For most apps I work with, this documentation must be honest.

The anti-pattern is to hide them using `@doc false` because these macros can not tell what functions are available, so people generate the docs and see the public API and the type spec and help Dialyzer-like tools.

You can use the docs to copy-paste things or narrow the app's focus to a particular module since they are juggling too many things at once. I know the typespec and, hopefully, some sample code is crucial. Especially as the application grows and you have less time to Memorize things.

So I would stick to documenting as much as possible, given that they are public APIs.

Reply:
> For most apps I work with, this documentation must be honest.

I'm not suggesting the documentation shouldn't be honest or really `@doc false`, but a thin pointer to the actual function as you've done below.

Reply:
Just rereading this, what was I thinking using the word "honest" I guess I meant "Hosted"... 🤦🏻 

Regardless, let me work on it!